WikiWatershed/model-my-watershed
# Add DEP Layers

#### Overview
To support a use case for municipalities in PA to conform to DEP reporting guidelines, we've added two new reference layers to the map.  Within PA, Municipalities and Urban Areas can be toggled on simultaneously to guide users in delineating a watershed with certain requirements within their municipality.  The ultimate workflow intends for the user to then generate a GMS file from the app which will be used in a special desktop version of GWLF-E with these specific reporting capabilities built in.  

Commits include:
- Updated data loading script to access sql files on s3
- Windshaft tile rendering
- Layers added to layer control
- Layer availability restriction to PA

See commit messages for details.

Connects #1388 
Connects #1389 
Connects #1390 

#### Testing
- Prior to testing, ensure your app vm is recently provisioned and bundle your scripts `./scripts/bundle.sh --debug`
- On your app server, run `setupdb.sh -p` to load the new DEP layers from s3 into your local database
  - If you do not have aws-cli installed, ignore the error at the end of the script, which is trying to purge the cache for this layer
  - Verify that the two tables are in your database, with spatial indexes, in `psql`
    - `\d dep_municipalities`
    - `\d dep_urban_areas`
- Open app and pan to an area within PA, verify that two new layers are enabled DEP Muncipalities (boundary) and DEP Urban Areas (coverage)
- Zoom away from PA so it is not in your viewport, verify that those two layers are disabled.
- Zoom back to PA, and turn on both layers.  Verify that that they render and that the boundary is rendered above the coverage, despite the order in which they are added
- Verify that Urban Areas has an opacity slider

![screenshot 2016-07-19 at 1 56 11 pm](https://cloud.githubusercontent.com/assets/1014341/16960693/93d63e8c-4db8-11e6-80fa-ab1d83cfc006.png)

Simple PA perimeter used (drawn from geojson.io)
![screenshot 2016-07-19 at 2 08 07 pm](https://cloud.githubusercontent.com/assets/1014341/16961151/4361c154-4dba-11e6-9e0d-f34f2f1617a1.png)
